docs/reference/contributing/guidelines/workflow.md

```diff
@@ -43,44 +43,37 @@ We copy issues reported on GitHub to our internal tracker and regularly triage t
-- You should always write commits to allow publication, so they can never contain confidential information, reference private documents, links to intranet locations or rude language.
+- Commits must always contain a proper description of their content. Start with a concise and sensible one-line description. Then, elaborate on reasoning of the choices made, descriptions for reviewers and other information that might otherwise be lost.
+- You should always write commits that allow publication. They must not contain confidential information, references to private documents, links to intranet locations or rude language.
 - Each commit should be the minimum self-contained commit for a change. A commit should always result in a new state that is again in a compilable state. You should (if possible) split large changes into logical smaller commits that help reviewers follow the reasoning behind the full change.
-- Commits and pull request titles should follow [Chris Beam’s seven rules of great commit messages](http://chris.beams.io/posts/git-commit#seven-rules):
-	1. Separate the subject from the body with a blank line.
-	1. Limit the subject line to 72 characters (note that this is a deviation from Beam's standard).
-	1. Capitalize the subject line.
-	1. Do not end the subject line with a period.
-	1. Use the imperative mood in the subject line.
-	1. Wrap the body at 72 characters.
-	1. Use the body to explain _what_ and _why_ vs _how_.
+- Commits and pull request titles should follow [Chris Beam’s seven rules of great commit messages](http://chris.beams.io/posts/git-commit#seven-rules).
 - Because we use GitHub, special commit tags that other projects may use, such as “Reviewed-by”, or “Signed-off-by”, are redundant and should be omitted. GitHub tracks who reviewed what and when.
 - Prefixing your commit message with a domain is acceptable, and we recommend doing so when it makes sense. However, prefixing the domain with the name of the repo is not useful. For example, making a commit entitled "mbed-drivers: Fix doppelwidget frobulation" to the `mbed-drivers` repo is not acceptable because it is already understood that the commit applies to `mbed-drivers`. Renaming the commit to "doppelwidget: Fix frobulation" would be better, if we presume that "doppelwidget" is a meaningful domain for changes, because it communicates that the change applies to the doppelwidget area of `mbed-drivers`.
 - All new features and enhancements require documentation, tests and user guides for us to accept them. Please link each pull request to all relevant documentation and test pull requests.
-- Avoid merging commmits. (Always rebase when possible.)
+- Avoid merging commmits. (Always rebase when possible.)<!-- Should we be stricter here? -> Do not merge your commits. -->
```

Comment:
@0xc0170 What are your thoughts? Did we word it this way for a reason, or is it OK to be stricter?

Reply:
ping @0xc0170